internal/metrics/metrics.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package metrics
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/open-policy-agent/opa/internal/metrics/prometheus"
+)
+
+// Metrics is the current global metrics provider
+var Metrics Provider
+
+func init() {
+	Metrics = &dummyProvider{}
+}
+
+// Provider is the API of metric providers which must be implemented by each provider
+type Provider interface {
+	RegisterEndpoints(registrar func(path, method string, handler http.Handler))
+	InstrumentHandler(handler http.Handler, label string) http.Handler
+	Gather() (interface{}, error)
+	Name() string
+}
+
+// SetProvider sets the current global metrics provider singleton by the provider name and its config settings
+func SetProvider(name string, config json.RawMessage) error {
```

Comment:
Per comment above if we inject the provider then we can remove this.

Reply:
Can you explain how? SetProvider sets the current, active metrics provider. Even if providers would register themselves how it changes which provider is the current?